### DISCUSSION TITLE
Data Received Event?

### DISCUSSION THREAD

**Discussion by sdeluca** [badcel/HidApi.Net#32] [Q&A] [answered]
Is it possible to use this library to subscribe to an event when data is received at the device?
I'm hoping that some equivalent of .Net's SerialPort.DataReceived event might be possible.

**Comment by badcel** [comment, marked as answer]
This library only wraps the available HidApi api which does not offer any events.

But you can probably set a Device into [non blocking mode](https://github.com/badcel/HidApi.Net/blob/65897cfb4dfef45cdfb8fe7a7de74e656b7c5d39/src/HidApi.Net/Device.cs#L108) and asynchronously check for new data. If new data is received raise an event.

HidApi docs: https://github.com/libusb/hidapi/blob/b516e970480a8d05b1c392d106ca110a9d06d668/hidapi/hidapi.h#LL372C13-L372C13

If you experience problems with the library let me know. Any feedback is welcome.

**Reply by sdeluca** [reply]
Aha ok, I see what you mean. I am new to all this HidApi stuff but will try implementing as you suggest.
Thank you.
